debug/testlib.py

```diff
@@ -383,11 +387,16 @@ def start(self, cmd, logfile, extra_env):
             while True:
                 m = self.expect(
                         rb"(Listening on port (\d+) for gdb connections|"
-                        rb"tcl server disabled)",
+                        rb"Listening on port (\d+) for tcl connections)",
                         message="Waiting for OpenOCD to start up...")
-                if b"gdb" in m.group(1):
+                if b"gdb connections" in m.group(1):
                     self.gdb_ports.append(int(m.group(2)))
-                else:
+                elif b"tcl connections" in m.group(1):
+                    if self.tclrpc_port:
+                      raise TestLibError("unexpected re-definition of TCL-RPC port")
+                    self.tclrpc_port = int(m.group(3))
+
+                if self.tclrpc_port and (len(self.gdb_ports) > 0):
```

Comment:
The issue is not introduced by this patch, but I think it worth pointing out. This condition relies on all GDB ports being reported by OpenOCD prior to TCL-RPC port.
In turn this depends on whether the call to `init` is implicit or explicit:
```
> openocd ... -c "gdb_port 0" -c "tcl_port 0"  -c "telnet_port disabled" |& grep 'Listening'
Info : Listening on port 41587 for tcl connections
Info : Listening on port 40911 for gdb connections
Info : Listening on port 33639 for gdb connections
```
```
> openocd ... -c "gdb_port 0" -c "tcl_port 0"  -c "telnet_port disabled" -c init |& grep 'Listening'
Info : Listening on port 40575 for gdb connections
Info : Listening on port 38511 for gdb connections
Info : Listening on port 35989 for tcl connections
```
Currently, this works, since all the configs in `debug/targets/RISC-V` call `init` explicitly.
Wouldn't it be better to add some comment regarding this behavior?

Reply:
Let's not add anything unnecessary to this patch. It's fairly urgent since without this fix, OpenOCD development is hamstrung. It's better to merge this soon than it is to disable the tests while we add more improvements to this PR.

Reply:
I've already addressed this. Just added a comment